osm2gtfs/creators/trips_creator.py

```diff
@@ -13,32 +19,301 @@ def __repr__(self):
                 shape.AddPoint(
                     lat=float(point["lat"]), lon=float(point["lon"]))
             feed.AddShapeObject(shape)
         return shape_id
+
+    def _add_itinerary_trips(self, feed, itinerary, line, trip_builder,
+                             shape_id):
+        """
+        Add all trips of an itinerary to the GTFS feed.
+        """
+        # Obtain GTFS route to add trips to it.
+        route = feed.GetRoute(line.route_id)
+        trips_count = 0
+
+        # Loop through each timeslot for a trip
+        for trip in trip_builder['schedule']:
+            gtfs_trip = route.AddTrip(feed, headsign=itinerary.to,
+                                      service_period=trip_builder['service_period'])
+            trips_count += 1
+
+            # Go through all stops of an itinerary
+            for itinerary_stop_id in itinerary.get_stops():
+
+                # Load full stop object
+                try:
+                    itinerary_stop = trip_builder[
+                        'all_stops']['regular'][itinerary_stop_id]
+                except ValueError:
+                    print("Itinerary (" + itinerary.route_url +
+                          ") misses a stop:")
+                    print(" Please review:" + itinerary_stop.osm_url)
+                    continue
+
+                try:
+                    # Load respective GTFS stop object
+                    gtfs_stop = feed.GetStop(str(itinerary_stop.stop_id))
+                except ValueError:
+                    print("Warning: Stop in itinerary was not found in GTFS.")
+                    print(itinerary_stop.osm_url)
+
+                # Make sure we compare same unicode encoding
+                if type(itinerary_stop.name) is str:
+                    itinerary_stop.name = itinerary_stop.name.decode('utf-8')
+
+                time = "-"
+                # Check if we have specific time information for this stop.
+                try:
+                    time = trip[trip_builder['stops'].index(itinerary_stop.name)]
+                except ValueError:
+                    pass
+
+                # Validate time information
+                if time != "-":
+                    try:
+                        time_at_stop = str(
+                            datetime.strptime(time, "%H:%M").time())
+                    except ValueError:
+                        print("Warning: Time for a stop was not valid.")
+                        print(" " + itinerary_stop.name +
+                              " - " + itinerary_stop.osm_id)
+                        break
+                    gtfs_trip.AddStopTime(gtfs_stop, stop_time=time_at_stop)
+
+                # Add stop without time information, too (we interpolate later)
+                else:
+                    try:
+                        gtfs_trip.AddStopTime(gtfs_stop)
+                    except ValueError:
+                        print("Warning: Could not add a stop to trip.")
```

Comment:
If I remember correctly *transitfeed* only throws this `ValueError` when you try to add the first stop of a trip without time. We could adjust the message to include that information.